backend/msgpack/conn_test.go

```diff
@@ -22,78 +22,149 @@ package msgpack
 
 import (
 	"errors"
+	"fmt"
+	"math"
 	"net"
 	"sync"
 	"testing"
 	"time"
 
 	"github.com/m3db/m3x/clock"
 
+	"github.com/leanovate/gopter"
+	"github.com/leanovate/gopter/gen"
+	"github.com/leanovate/gopter/prop"
 	"github.com/stretchr/testify/require"
 )
 
 const (
-	testFakeServerAddr  = "nonexistent"
-	testLocalServerAddr = "127.0.0.1:0"
+	testFakeServerAddr        = "nonexistent"
+	testLocalServerAddr       = "127.0.0.1:0"
+	testRandomSeeed           = 831992
+	testMinSuccessfulTests    = 1000
+	testReconnectThreshold    = 1024
+	testMaxReconnectThreshold = 8096
 )
 
 var (
 	errTestConnect = errors.New("connect error")
 	errTestWrite   = errors.New("write error")
 )
 
-func TestConnectionWriteNoReconnect(t *testing.T) {
-	conn := newConnection(testFakeServerAddr, testConnectionOptions())
-	conn.connectWithLockFn = func() error { return errTestConnect }
+func TestConnectionProperties(t *testing.T) {
+	params := gopter.DefaultTestParameters()
+	params.Rng.Seed(testRandomSeeed)
+	params.MinSuccessfulTests = testMinSuccessfulTests
+	props := gopter.NewProperties(params)
 
-	require.Equal(t, errNoActiveConnection, conn.Write(nil))
-	require.Equal(t, 1, conn.numFailures)
-}
+	props.Property(
```

Comment:
Tests look great!

minor suggestion (take/leave): break each property into a separate test fn. will make it easier to repro any issues in isolation

Reply:
Sounds good, yea I wasn't sure what the best practice here was, the fact that one could add more properties to `props` made me think that maybe it would make sense to group them together, but being able to run in isolation definitely seems more worthwhile.

Reply:
Cool, good stuff.